### PR TITLE
[OSCD] Detects Obfuscated Powershell via ENCODING OBFUSCATION\8 #18 (4104, 4103 + Services + process_creation)

### PR DESCRIPTION
**Pull request by zinint**
#1009 task 18

### CODE REVIEW

**Comment by yugoslavskiy** [comment]
Hello @thomaspatzke! We have pretty much the same issue here, as in the https://github.com/Neo23x0/sigma/pull/1225 

Could you please give us a hand?
Thank you!

File: rules/windows/builtin/win_invoke_obfuscation_via_enc_obfuscation_8_services.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    - attack.defense_evasion
+    - attack.t1027
+    - attack.execution
+    - attack.t1059.001
+falsepositives:
+    - Unknown
+level: medium
+detection:
+    selection_1:
+        - ImagePath|re: '(?i)\'\s+\'\s?\|(?:foreach|\s?%).*(?:\[string\]::join|remove\.tostring).*trim.*split.*(?:char|substring\.tostring).*\}'
```
Comment (yugoslavskiy) [inline rules/windows/builtin/win_invoke_obfuscation_via_enc_obfuscation_8_services.yml:20]: maybe this construction is going to fix that?

```
        ImagePath|re: >
          '(?i)\'\s+\'\s?\|(?:foreach|\s?%).*(?:\[string\]::join|remove\.tostring).*trim.*split.*(?:char|substring\.tostring).*\}'
```

It works locally. Here is the result:

```
(EventID:"7045" AND ImagePath:/'(?i)\'\s+\'\s?\|(?:foreach|\s?%).*(?:\[string\]::join|remove\.tostring).*trim.*split.*(?:char|substring\.tostring).*\}'
/)
(EventLog:"Microsoft\-Windows\-Sysmon" AND EventID:"6" AND ImagePath:/'(?i)\'\s+\'\s?\|(?:foreach|\s?%).*(?:\[string\]::join|remove\.tostring).*trim.*split.*(?:char|substring\.tostring).*\}'
/)
(EventLog:"Security" AND EventID:"4697" AND ImagePath:/'(?i)\'\s+\'\s?\|(?:foreach|\s?%).*(?:\[string\]::join|remove\.tostring).*trim.*split.*(?:char|substring\.tostring).*\}'
/)

```

**Comment by zinint** [comment]
Hey, thanks @yugoslavskiy I'll try this and will get back to you!

**Comment by zinint** [comment]
@yugoslavskiy thanks a lot for your help! It worked! Really appreciate it!

I'm passing the test `Sigma Tools and Rules` right now, but now I'm stuck at this one `Test Generated Elasticsearch Query Strings` with the following error:
``` python
!!! Query check failed!
Unclosed client session
client_session: <aiohttp.client.ClientSession object at 0x7f764a991040>
Unclosed connector
connections: ['[(<aiohttp.client_proto.ResponseHandler object at 0x7f7649ef44c0>, 984.354748878)]']
connector: <aiohttp.connector.TCPConnector object at 0x7f764a988fd0>
make: *** [test-backend-es-qs] Error 3
Makefile:115: recipe for target 'test-backend-es-qs' failed
Error: Process completed with exit code 2.
```
is this a backend error?

**Comment by thomaspatzke** [comment]
Sorry for that. Unfortunately the ES query test stopped to work ~2 weeks ago and so all checks fail on PRs which are branched from master in the state with enabled check. The check is now disabled and check pass now again. Please try to rebase you PR to the new master or cherry pick the commit [31241d9bb](https://github.com/Neo23x0/sigma/commit/31241d9bbdde78575293ec97da83e193b902b47c).

File: rules/windows/builtin/win_invoke_obfuscation_via_enc_obfuscation_8_services.yml (hunk 1 above)
Comment (yugoslavskiy) [reply rules/windows/builtin/win_invoke_obfuscation_via_enc_obfuscation_8_services.yml]: Had to add `-` to the `ImagePath|re: >` to strip the line. my fault. I've just went through all your rules and updated them

**Comment by zinint** [comment]
Oops (: smth has gone wrong (:

**Comment by zinint** [comment]
Well, don't try to do this from GUI (: Use CLI (:

**Comment by zinint** [comment]
@yugoslavskiy, I believe that now it's ready to be merged (:

**Comment by yugoslavskiy** [comment]
Hey @zinint! 
Could you please rebase your PR so it would include only your rules?
There are lots of other files modified and it quite hard to find your rules for the review.
As you know, it's not the last stage of the review, and it will save lots of time for all of us. 
Thanks, man!

**Pull request closed without merging**